victim-loss-identifier/src/agent.ts

```diff
@@ -1,16 +1,171 @@
+              scammerAddress,
+              NINETY_DAYS,
+              dataFetcher,
+              scammersCurrentlyMonitored,
+              victimsScammed
+            ))
+          );
+          break;
+
+        default:
+          return findings;
+      }
+    }
+
+    return findings;
+  };
+}
+
+export function provideHandleBlock(): HandleBlock {
+  return async (blockEvent: BlockEvent): Promise<Finding[]> => {
+    const findings: Finding[] = [];
+
+    const blocksInOneDay = getBlocksInTimePeriodForChainId(ONE_DAY_IN_SECS, chainId);
+
+    if (blockEvent.blockNumber % blocksInOneDay === 0) {
+      for (const scammerAddress of Object.keys(scammersCurrentlyMonitored)) {
+        const blocksSinceScammerLastActive =
+          blockEvent.blockNumber - scammersCurrentlyMonitored[scammerAddress].mostRecentActivityByBlockNumber;
+
+        const scammerLastActiveInSecs = blocksSinceScammerLastActive * getChainBlockTime(chainId);
+        const daysSinceScammerLastActive = Math.ceil(scammerLastActiveInSecs / ONE_DAY_IN_SECS);
+
+        const fraudulentNftOrderFindings = await processFraudulentNftOrders(
+          scammerAddress,
+          daysSinceScammerLastActive,
+          dataFetcher,
+          scammersCurrentlyMonitored,
+          victimsScammed
+        );
+
+        if (fraudulentNftOrderFindings.length > 0) {
+          scammersCurrentlyMonitored[scammerAddress].mostRecentActivityByBlockNumber = blockEvent.blockNumber;
+        }
+
+        findings.push(...fraudulentNftOrderFindings);
+
+        const blocksInThirtyDays = getBlocksInTimePeriodForChainId(THIRTY_DAYS_IN_SECS, chainId);
+        if (
+          blockEvent.blockNumber - scammersCurrentlyMonitored[scammerAddress].mostRecentActivityByBlockNumber >
+          blocksInThirtyDays
+        ) {
+          delete scammersCurrentlyMonitored[scammerAddress];
+        }
+      }
+    }
+
+    const date = new Date();
+    const minutes = date.getMinutes();
+
+    if (minutes % 10 === 0 && lastPersistenceMinute !== minutes) {
+      let scammersObjectSize = Buffer.from(JSON.stringify(scammersCurrentlyMonitored)).length;
+      let victimsObjectSize = Buffer.from(JSON.stringify(victimsScammed)).length;
+      console.log(
+        `Scammers Monitored Object Size: ${scammersObjectSize} | Victims Scammed Object Size: ${victimsObjectSize}`
+      );
+
+      let largerObjectSize = scammersObjectSize < victimsObjectSize ? victimsObjectSize : scammersObjectSize;
+
+      while (largerObjectSize > MAX_OBJECT_SIZE) {
+        console.log("Cleaning Scammers Monitored Object of size: ", scammersObjectSize);
+        cleanObject(scammersCurrentlyMonitored);
+        scammersObjectSize = Buffer.from(JSON.stringify(scammersCurrentlyMonitored)).length;
+        console.log("Scammers Monitored Object after cleaning: ", scammersObjectSize);
+
+        console.log("Cleaning Victims Scammed Object of size: ", victimsObjectSize);
+        cleanObject(victimsScammed);
```

Comment:
I think this may not work as expected, as inside the `cleanObject` we're deleting 1/4 of the object (oldest) entries and these entries may not be the same for both objects.

I believe a better solution is to pass the largest object inside the `cleanObject` and inside that function, delete the corresponding entries from the other (not-largest) object.

WDYT?

Reply:
Thinking about it, I'm not sure _this_ new approach would be ideal either. That's because the victim's now have their own `mostRecentActivityByBlockNumber`. So if they were scammed by a different scammer, the victim would have been active due to that.

For example, say `scammerOne` scammed `victimOne` and `scammerOne` became inactive afterwards. _Then_ `scammerTwo` scammed `victimOne` a good amount of time after `scammerOne` scammed `victimOne`.

If the time came to clean out the objects, if `scammerOne` was eligible to be removed, it would then be incorrect to remove `victimOne` because they were still "active" after having been scammed by `scammerOne` when they were scammed by `scammerTwo`.

So in the end, I don't think the entries being deleted for both _need_ to be the same. If we look at a scammer, if a specific victim was removed, since they would have already been accounted for, I don't know that it matters. The inverse is even more true because a victim could be scammed by a subsequent scammer and thus needs to be counted as part of _that_ scam.

WDYT?